sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNodeSuite.scala

```diff
@@ -72,11 +72,39 @@ class SqlScriptingExecutionNodeSuite extends SparkFunSuite with SharedSparkSessi
     }
   }
 
+  case class TestRepeatCondition(
+    condVal: Boolean, reps: Int, description: String)
+    extends SingleStatementExec(
+      parsedPlan = DummyLogicalPlan(),
+      Origin(startIndex = Some(0), stopIndex = Some(description.length)),
+      isInternal = false)
+
+  case class TestRepeat(
+    condition: TestRepeatCondition,
+    body: CompoundBodyExec)
+    extends RepeatStatementExec(condition, body, spark) {
+
+    private var callCount: Int = 0
+
+    override def evaluateBooleanCondition(
```

Comment:
same here, it's the same as in While loop - let's create one function/lambda outside of the classes and use it here